### DISCUSSION TITLE
Weird problem when calling tkinter.filedialog.askdirectory() by VBtn.Click

### DISCUSSION THREAD

**Discussion by Brandon-Xu** [Kitware/trame#457] [General] [answered]
I want to get the folder path selected by the user by calling this function **tkinter.filedialog.askdirectory()**
But I must call it once before **server.start()**
otherwise this function will only execute the first line of **print("before")** and subsequent code will not be executed.

```
from trame.app import get_server
from trame.ui.vuetify3 import SinglePageLayout
from trame.widgets import vuetify3
from tkinter import filedialog

server = get_server(client_type="vue3")


def ask_directory():
    print("before")
    path = filedialog.askdirectory()
    print("folder path is:", path)


with SinglePageLayout(server) as layout:
    layout.title.set_text("Hello trame")
    with layout.content:
        vuetify3.VBtn("Ask Directory", click=ask_directory)


# ask_directory() #  I must call it once before server.start() ┑(￣Д ￣)┍

if __name__ == "__main__":
    server.start()

```

any one can help help me ?

**Comment by jourdain** [comment]
`server.controller.on_server_ready.add(ask_directory)`

**Reply by Brandon-Xu** [reply, marked as answer]
sorry, I didn't describe it clearly enough, I want this function to be called by clicking the button
but I solved the problem myself, In fact, the window was opened successfully, 
but it was covered under other program windows, which made me think it was not successful.

This is my improved code

```
_root = tkinter.Tk()
_root.withdraw()

def ask_directory():
    _root.lift()
    _root.attributes("-topmost", True)
    directory = filedialog.askdirectory()
    if directory:
        return directory
    else:
        return None
```
